src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -47,17 +51,53 @@ public void Start(PushRuntimeSettings limitations)
 
             if (purgeOnStartup)
             {
-                Array.ForEach(Directory.GetFiles(path), File.Delete);
+                PurgeDirectories();
             }
 
+            CreateDirectories();
+
+            RecoverPendingTransactions();
```

Comment:
just to confirm, PurgeDirectories would also delete remaining transaction directories, right?